In [1]:
library(tidyverse)
library(dplyr)
library(repr)
library(tidymodels)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.4     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
── Attaching packages ────────────────────────────────────── tidymodels 1.1.1 ──

✔ broom        1.0.5     ✔ rsample      1.2.0
✔ dials        1.2.0     ✔ tune         1.1.2
✔ infer        1.0.5     ✔ workflows    1.1.3
✔ modeldata    1.2.0     ✔ workflowsets 1.0.1
✔ parsnip      1.1.1     ✔ yardstick    1.2.0
✔ recipes      1.0.8     

── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ scales::discard() masks purrr::discard()
✖ dplyr::filt

# GROUP PROPOSAL: WINE TASTING

## Introduction

## Preliminary exploratory data analysis

1. Data Reading

In [2]:
temp <- tempfile()
download_file <- download.file("https://archive.ics.uci.edu/static/public/186/wine+quality.zip", temp)
red_wine <- read.csv2(unz(temp, "winequality-red.csv")) |>
             mutate(type = "Red")
white_wine <- read.csv2(unz(temp, "winequality-white.csv")) |>
              mutate(type ="White")
unlink(temp)

2. Data Wrangling

In [7]:
wine_data <- rbind(red_wine, white_wine) |>
             mutate(across(fixed.acidity:alcohol, as.numeric), quality = as_factor(quality),
                   type = as_factor(type))
write_csv(wine_data, "../data/wine_data.csv")
head(wine_data)

fixed.acidity volatile.acidity citric.acid residual.sugar chlorides
1  7.4          0.70             0.00        1.9            0.076    
2  7.8          0.88             0.00        2.6            0.098    
3  7.8          0.76             0.04        2.3            0.092    
4 11.2          0.28             0.56        1.9            0.075    
5  7.4          0.70             0.00        1.9            0.076    
6  7.4          0.66             0.00        1.8            0.075    
  free.sulfur.dioxide total.sulfur.dioxide density pH   sulphates alcohol
1 11                  34                   0.9978  3.51 0.56      9.4    
2 25                  67                   0.9968  3.20 0.68      9.8    
3 15                  54                   0.9970  3.26 0.65      9.8    
4 17                  60                   0.9980  3.16 0.58      9.8    
5 11                  34                   0.9978  3.51 0.56      9.4    
6 13                  40                   0.9978  3.51 0.56      9.4    
  quality type
1 5       Red 
2 5       Red 
3 5       Red 
4 6       Red 
5 5       Red 
6 5       Red

3. Summarization

a. The number of observations in each red wine/white wine class

In [8]:
wine_type_number <- wine_data |>
                    group_by(type) |>
                    summarize(Number = n())
wine_type_number

type  Number
1 Red   1599  
2 White 4898

Therefore, we have 4898 observations for White wine and 1599 observations for Red wine.

b. The number of observations for each quality level of White and Red wine.

In [10]:
quality_number <- wine_data |>
                  group_by(type, quality) |>
                  summarize(Number = n())
quality_number

`summarise()` has grouped output by 'type'. You can override using the
`.groups` argument.


type  quality Number
1  Red   3         10  
2  Red   4         53  
3  Red   5        681  
4  Red   6        638  
5  Red   7        199  
6  Red   8         18  
7  White 3         20  
8  White 4        163  
9  White 5       1457  
10 White 6       2198  
11 White 7        880  
12 White 8        175  
13 White 9          5

## Methods

## Expected outcomes and significance